src/basics.md

```diff
@@ -411,6 +412,45 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-simple-error-handling]: #ex-simple-error-handling
+<a name="ex-simple-error-handling"></a>
+## Handle errors correctly in main
+
+[![error-chain-badge]][error-chain] [![cat-rust-patterns-badge]][cat-rust-patterns]
+
+Handles error that occur when we try to open a file that does not exist. We do this using [error-chain], a library that takes care of a lot of boilerplate code needed in order to [handle errors in Rust].
+
+We call `Io(std::io::Error)` inside `foreign_links` to convert the type `std::io::Error` into error-chain's custom version of it.
+
+To handle the result of a function that might return an error, in this example `open_foo`, it must return a `Result`. To handle that result, we use pattern matching on it when calling the function. In this example we print an error message if that result was an error.
```

Comment:
I would avid repetition `To handle the result,` and `To handle that result,`. This might be rephrased and further simplified